plugin-server/tests/shared/process-event.ts

```diff
@@ -2162,5 +2166,128 @@ export const createProcessEventTests = (
         })
     }
 
+    test('set and set_once on the same key', async () => {
+        await createPerson(hub, team, ['distinct_id1'])
+
+        await processEvent(
+            'distinct_id1',
+            '',
+            '',
+            {
+                event: 'some_event',
+                properties: {
+                    token: team.api_token,
+                    distinct_id: 'distinct_id1',
+                    $set: { a_prop: 'test-set' },
+                    $set_once: { a_prop: 'test-set_once' },
+                },
+            } as any as PluginEvent,
+            team.id,
+            now,
+            now,
+            new UUIDT().toString()
+        )
+        expect((await hub.db.fetchEvents()).length).toBe(1)
+
+        const [event] = await hub.db.fetchEvents()
+        expect(event.properties['$set']).toEqual({ a_prop: 'test-set' })
+        expect(event.properties['$set_once']).toEqual({ a_prop: 'test-set_once' })
+
+        const [person] = await hub.db.fetchPersons()
+        expect(await hub.db.fetchDistinctIdValues(person)).toEqual(['distinct_id1'])
+        expect(person.properties).toEqual({ a_prop: 'test-set' })
+    })
+
+    describe('ingestion in any order', () => {
```

Comment:
won't these fail on one of the runs?

Reply:
I have the if case later `if (includeNewPropertiesUpdatesTests) {` for the tests that are not in order, but wanted to include the first test for the current set/set_once updates